### PR TITLE
[supply] Support default changelog

### PR DESCRIPTION
**Pull request by janicduplessis**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Currently on Android using supply it is needed to create a release notes file for every build. Most release end up including the same release notes so having a way to specify default notes if none are provided can save a lot of time and prevent release without notes.

### Description
This adds support for default release notes when a release note file for the current build does not exist.

It works by looking for a file named `default.txt` if a file with the build number does not exist.

### Testing Steps

Tested in my project by patching fastlane locally and adding a `default.txt` file with some text under the `changelogs` directory. Then I ran my deployment script and checked that the text included in `default.txt` was set as release notes in the google play console.

### CODE REVIEW

File: supply/lib/supply/uploader.rb [hunk 1]
```diff
@@ -195,7 +195,13 @@ def upload_changelog(language, version_code)
         UI.message("Updating changelog for '#{version_code}' and language '#{language}'...")
         changelog_text = File.read(path, encoding: 'UTF-8')
       else
-        UI.message("Could not find changelog for '#{version_code}' and language '#{language}' at path #{path}...")
+        path = File.join(Supply.config[:metadata_path], language, Supply::CHANGELOGS_FOLDER_NAME, "default.txt")
+        if File.exist?(path)
+          UI.message("Updating default changelog for '#{version_code}' and language '#{language}'...")
```
Comment (janpio) [inline supply/lib/supply/uploader.rb]:
```suggestion
          UI.message("Updating changelog for '#{version_code}' and language '#{language}' to default changelog...")
```

File: supply/lib/supply/uploader.rb [hunk 2]
```diff
@@ -195,7 +195,13 @@ def upload_changelog(language, version_code)
         UI.message("Updating changelog for '#{version_code}' and language '#{language}'...")
         changelog_text = File.read(path, encoding: 'UTF-8')
       else
-        UI.message("Could not find changelog for '#{version_code}' and language '#{language}' at path #{path}...")
+        path = File.join(Supply.config[:metadata_path], language, Supply::CHANGELOGS_FOLDER_NAME, "default.txt")
```
Comment (janpio) [inline supply/lib/supply/uploader.rb]: rename to `default_changelog_path` maybe

**Comment by janpio** [comment]
Thanks, this looks great @janicduplessis and is a great idea. Left 2 comments with nitpicks.

File: supply/lib/supply/uploader.rb (hunk 1 above)
Comment (janicduplessis) [reply supply/lib/supply/uploader.rb]: I wanted I slightly different message so you can know whether it used the default changelog or not, what do you think about keeping that?

Comment (janpio) [reply supply/lib/supply/uploader.rb]: Yeah, but what happens here is that the "changelog" is updated. In this new case it is "update to the default changelog text" - hence my suggestion.

Comment (janicduplessis) [reply supply/lib/supply/uploader.rb]: Oh my bad I didn’t notice the text added at the end of the line! Looks good.

**Review by joshdholtz — APPROVED** [review]
This is amazing! Thanks for taking the time to add this 😊 Really appreciate the contribution ❤️

**Pull request merged**